Rakefile

```diff
@@ -51,7 +51,7 @@ end
 def defaults(args)
   bindir = args.bindir.nil? || args.bindir.empty? ? Pathname.new('./build/swiftgen/bin') : Pathname.new(args.bindir)
   fmkdir = args.fmkdir.nil? || args.fmkdir.empty? ? bindir + '../lib' : Pathname.new(args.fmkdir)
-  tpldir = args.tpldir.nil? || args.tpldir.empty? ? bindir + '../templates' : Pathname.new(args.tpldir)
+  tpldir = args.tpldir.nil? || args.tpldir.empty? ? bindir + '../Resources/templates' : Pathname.new(args.tpldir)
```

Comment:
Ah, I see.

But I think even without this PR, the `main.swift` was modified by the `Rakefile` anyway, as it can always be installed in any place you like depending on the rake task arguments…

And I'd still prefer keeping the structure of the final binary untouched, first because it's more standard, second because changing it would mean adapting the Homebrew's formula too, and because the final product is more important and the .app is just a convenience structure for development, not for distribution.

So I think the best option is to:

* Keep the templates in Resources for the development mode (.app bundle) — We could even point the `templatesRelativePath` to `NSBundle.main.resourcesDirectory` to make it more dynamic? — because that's what .app bundles expect and the place where Xcode will put them
* Keep the `Rakefile` changing that value to `$tpldir`, which by default if not specified when calling `rake`, would be `../templates` to match what Homebrew and CLI binaries expect
* _Don't_ `git reset main.swift` at the end of `rake`. The `Rakefile` already did change the `main.swift` without reverting it before that PR, and I think that's OK:
  * now that the project will be organized as an app bundle, we should only have to use the `Rakefile` to do a release, so not that often anyway
  * I prefer to have a dirty git working copy after the `rake install` and handle the reset manually than having `rake` do a `git checkout main.swift` to reset it… and potentially accidentally also revert other changes that we might have forgotten to commit before calling `rake install` (another option would be to ensure that everything has been committed to git before authorizing to continue to `rake install` but ¯\_(ツ)_/¯ )

So, in summary, commit the value of the `relativePath` that matches the relative path when the .app bundle stores the templates once compiled, so that a Cmd-R (and Cmd-U) works out of the box without changes from Xcode, and keep the Rakefile behavior as before, still letting it change the `templateRelativePath` during the `rake install` to match where the templates are asked to be installed according to the `rake install[…]` parameters, which would still default to `../templates`.